main/entry.c

```diff
@@ -195,10 +198,23 @@ extern void makeFileTag (const char *const fileName)
 		   unnecessary read line loop. */
 		while (readLineFromInputFile () != NULL)
 			; /* Do nothing */
-		tag.extensionFields.endLine = getInputLineNumber ();
+		endLine = getInputLineNumber ();
 	}
 
+	tag.extensionFields.endLine = endLine;
 	makeTagEntry (&tag);
+
+	/* add the full filename too */
```

Comment:
I think it is better to emit full filename only when --extras=+q is given.
So the comment will be :
```
/* add the full filename too if --extras=+q is given. */
```